gcsio/src/main/java/com/google/cloud/hadoop/gcsio/authorization/AuthorizationMode.java

```diff
@@ -0,0 +1,18 @@
+package com.google.cloud.hadoop.gcsio.authorization;
+
+/**
+ * The Authorization mode defines how to obtain the Access Token.
+ * GENERIC means the generated Access Token works for any GCS request.
+ * REQUEST_CONTEXT_RELATED means the generated Access Token works for the objects/actions
+ * specified in the request
+ */
+public enum AuthorizationMode {
+  GENERIC("GENERIC"),
+  REQUEST_CONTEXT_RELATED("REQUEST_CONTEXT_RELATED");
+
+  private final String authorizationTypeName;
```

Comment:
Why do we need a name?

Reply:
Just keeping the door open to evolve this enum when we refactor/consolidate all Authorization codes (i.e. each mode will have certain flags set/turned off. Instead of having flags all over the place). Let me know if you feel strongly I can keep it simple for now and let the refactoring decide what should happen.